spec/graphql/query_spec.rb

```diff
@@ -124,6 +124,20 @@
     end
   end
 
+  it "fails to execute a query containing a type definition" do
+    query_string = '
+      { root }
+
+      type Query { foo: String }
+    '
+    begin
+      GraphQL::Query.new(schema, query_string)
+      flunk "Expected an exception"
+    rescue GraphQL::ExecutionError => exc
+      assert_equal "GraphQL query cannot contain a schema definition", exc.message
+    end
```

Comment:
could we use `assert_raises` here? for example 

```ruby 
exc = assert_raises(GraphQL::ExecutionError) { GraphQL::Query.new(schema, query_string) } 
assert_equal "...", exc.message
```

?

If that works, I prefer it because of the `assert_` language!